apps/developer-portal/src/components/applications/forms/inbound-oidc-form.tsx

```diff
@@ -503,6 +522,10 @@ export const InboundOIDCForm: FunctionComponent<InboundOIDCFormPropsInterface> =
                             addURLTooltip={ t("common:addURL") }
                             duplicateURLErrorMessage={ t("common:duplicateURLError") }
                             data-testid={ `${ testId }-callback-url-input` }
+                            getSubmitAndInputValue={ (callSubmit: () => string, url: () => string) => {
+                                submitUrl = callSubmit;
```

Comment:
Is `callSubmit` gives the final set of URLs for submitting?

Reply:
No, this triggers the `addUrl` method inside the `URLInput` component

Reply:
IMO, a trigger returning a value is a bit unnatural. Can't we think of a different way to implement the behaviour. Also It would be better if we could avoid reassigning the callbacks to local functions.

Reply:
We need to have a two-way communication between the parent component and the `URLInput` component. Once you trigger the `addUrl` method, the parent component should be able to know if there are any validation issues. Another way of doing this would be to move the state of the `URLInput` component entirely to a context or redux.